### ISSUE TITLE
Add languages

### ISSUE THREAD

**Issue by nozem79** [naggety/picotts#7]
I do not know how to properly ask a question in Github, so I hope this is the right way.
Since I presume the voices are based on WAV's how difficult is it to ad a language like dutch & what is te proper way to start do do that?

Kind regardsm

Daniel

**Comment by jpwhiting** [comment]
Hello Daniel,

Actually, the languages are based on some data files that are in a format that is known, but generated by some other tools that aren't included in this project. There are a dozen or so files, mostly lists of numbers for signal generation, phoneme pronunciation, etc. etc. Included in upstream git repository are windows only binary tools that convert those into data files that pico library can read and use. In this other fork I created here: https://github.com/jpwhiting/svox-pico I have tried to reverse engineer some of those tools and rewrite them in python so they will work on linux/mac also, but haven't got them all yet. Once those are all converted and rewritten we will be able to generate the existing data files for the existing languages. Understanding the source data files would be required to add more languages. I hope we get to that at some point, but I think it's going to be far off.

The other problem is that the company that created this (SVOX) has been bought by Google last decade so there's not really a known person or place to reach out to for guidance in figuring this out as far as I can tell.

**Comment by nozem79** [comment]
Thanks fot the swift replay, I hope it's possible to use Pico in my language of choice.

**Comment by ihuguet** [comment]
@jpwhiting, although having those tools as free software would be great, in the meanwhile those binaries can be used to generate the languages, is that correct?

What's upstream git repo? I just got this sources from debian's package.

**Comment by jpwhiting** [comment]
Yes, those windows tools work fine for now. I think they are preventing debian from putting pico in main though, they are in non-free because those exe files are an unknown I guess https://packages.debian.org/bullseye/libttspico0

Upstream repo for svox-pico is here: https://android.googlesource.com/platform/external/svox/ the pdfs documenting the data files are on donut-release branch iirc.

**Comment by nozem79** [comment]
Somehow I cannot find the exe's , then again, I'm half blind ;-)

**Comment by jpwhiting** [comment]
Daniel,

They are here:
https://android.googlesource.com/platform/external/svox/+/refs/heads/donut-release
But they only convert the textfiles to .pkb then there's a perl script to
combine those into .bin files 2 per language, one for the textanalysis bits
and the other for the sound generation bits. The trick to add a new
language is to come up with the text files used as the starting point for
generating each of those. That commit also has some useful pdfs about how
it all works, but there's not much about where the data files came from or
how to add new ones unfortunately.

BR,
Jeremy

**Comment by Pallas1303** [comment]
Hi, I was exploring some folder of my Samsung J7 Prime phone. Until I found two folders in /system/tts/ called lang_SMT and lang_SVOXP with several files .qs .am .cfg .dt .lng in lang_SMT and a svoxconfig.txt file in lang_SVOXP called svoxconfig.txt

**Comment by versae** [comment]
I was looking at the outputs of pico and they look great, but was wondering how to add other languages and then found this issue. Any luck at all adding a new language?

**Comment by Akszyn** [comment]
Hello, I am abiding to the question of whether there is a chance for more languages, e.g. Polish?

**Comment by bhcuong2008** [comment]
I go to this link but can't download documents. I need to add my custom language.
https://android.googlesource.com/platform/external/svox/+/refs/heads/donut-release
